tony-core/src/main/java/com/linkedin/tony/TonyApplicationMaster.java

```diff
@@ -386,6 +453,28 @@ private boolean prepare() {
       LOG.error("Exception while preparing AM", e);
       return false;
     }
+
+    // Write config file to HDFS
+    jobDir = new Path(tonyHistoryFolder + appIdString + "/");
+    try {
+      if (!fs.exists(jobDir)) {
+        fs.mkdirs(jobDir);
+        fs.setPermission(jobDir, new FsPermission((short) 0770));
```

Comment:
Can you move this history folder setup logic out into a separate method like `setupHistoryDirectory` for readability?

Reply:
Fixed: https://github.com/linkedin/TonY/pull/79/commits/e489758601fce6c775bd5d11e0f97289e0664a81